docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java

```diff
@@ -110,16 +111,32 @@ public DockerTemplateBase(String image,
 
     protected Object readResolve() {
         if (volumesFrom != null) {
-            setVolumesFrom2(new String[]{volumesFrom});
+            if (StringUtils.isNotBlank(volumesFrom)) {
+                setVolumesFrom2(new String[]{volumesFrom});
+            }
             volumesFrom = null;
         }
 
         return this;
     }
 
+    //TODO move/replace
     public static String[] splitAndFilterEmpty(String s, String separator) {
         List<String> list = Splitter.on(separator).omitEmptyStrings().splitToList(s);
-        return list.toArray(new String[0]);
+        return list.toArray(new String[list.size()]);
+    }
+
+    //TODO move/replace
+    public static String[] filterStringArray(String[] arr) {
+        final ArrayList<String> strings = new ArrayList<>();
```

Comment:
don't use concrete class names if it has interface - in this case it should be List